packages/design-system/src/stories/foundation/colors/TokenPaletteTable.tsx

```diff
@@ -1,28 +1,55 @@
 import React from "react";
-import { ComponentStory, ComponentMeta } from "@storybook/react";
 import {
-  useTheme,
   Box,
   Table,
   TableHead,
   TableRow,
   TableBody,
   TableContainer,
   TableCell,
+  useTheme,
 } from "@mui/material";
 import { StyledTypography } from "./styled";
 import { tokenCoreColor } from "@/foundation/colors/token/core";
+import { tokenComponentColor } from "@/foundation/colors/token/component";
+
 import type { ColorToken } from "@/foundation/colors/types";
+import type { TokenComponentColorValue } from "@/foundation/colors/token/component";
+import type { ColorTokenValueBySurface } from "@/foundation/colors/token/types";
+import { light } from "@mui/material/styles/createPalette";
+
+interface TokenPaletteTableProps {
+  token: keyof ColorToken;
+}
 
-const CoreTokenColors = () => {
+const TokenPaletteTable = ({ token }: TokenPaletteTableProps) => {
   const theme = useTheme();
 
-  const colorBoxStyle = {
+  const COLOR_SURFACE: ColorTokenValueBySurface = {
+    light1: "var(--grey-0)",
+    light2: "var(--grey-10)",
+    dark1: "var(--grey-90)",
+    dark2: "var(--grey-85)",
+    dark3: "var(--grey-80)",
+    dark4: "var(--grey-70)",
+  } as const;
+
+  const COLOR_BOX_STYLE = {
     borderTop: "1px solid",
     borderBottom: "1px solid",
     borderColor: "rgba(0, 0, 0, 0.12)",
     height: "12px",
-  };
+  } as const;
+
+  const colorTokenMap = Object.entries(
+    token === "core" ? theme.palette.token.core : theme.palette.token.component
+  );
+
+  const tokenComponentColorFlatten = [
+    ...Object.values(tokenComponentColor),
+  ].reduce((acc, cur) => {
```

Comment:
```suggestion
  const tokenComponentColorFlatten = Object.values(tokenComponentColor).reduce((acc, cur) => {
```
Object.values의 반환값도 새 배열을 반환해줍니다. 😁

Reply:
앗 맞네요. 다소 이상한 방식으로 썼는데, 더 나은 방향 말씀해주셔서 감사합니다~!